### ISSUE TITLE
Improve `Breakpoint` ergonomics?

### ISSUE THREAD

**Issue by brunnerh** [carbon-design-system/carbon-components-svelte#1069]
One of the main questions usually is: Is the viewport larger/smaller than *X*?
Currently this requires either explicitly listing all valid sizes or resolving the size from the read-only `breakpoints` prop to compare it.

e.g.

```svelte
<script>
	import { Breakpoint } from 'carbon-components-svelte';
	
	let size;
	let breakpoints = {};
	$: tooSmall = breakpoints[size] < breakpoints.lg;
</script>

<Breakpoint bind:size bind:breakpoints />
```

This is not great because:
1. The values are only available after the component is mounted.
2. The values are only available within components and not code in imported JS files (e.g. extracted layout logic).
3. Component is redundant because it does not do anything component-specific (like adding elements to the DOM).

At first I thought that maybe it should remain a component because of the event handler lifecycle management, but it can be converted to a store without sacrificing that: `onMount` and `onDestroy` can be used in external files, as long as the code is called from a component.

[Proof of concept](https://svelte.dev/repl/1cc1ff0aa32e457388149262f9453670?version=3.46.4) (Toggle checkbox and see console drawer for lifecycle outputs)

I would propose to convert the component into:
1. A function that creates a readable store that returns the current breakpoint size.
2. An export (e.g. `breakpoints`) that contains the object from the `breakpoints` prop.

This would allow changing the above example to something like this:

```svelte
<script>
	import { breakpointObserver, breakpoints } from 'carbon-components-svelte';

	const breakpoint = breakpointObserver();
	$: tooSmall = breakpoints[$breakpoint] < breakpoints.lg;
</script>
```

The returned store could also expose utility functions that create derived stores for checking smaller/larger. This would simplify the code even further:

```svelte
<script>
	import { breakpointObserver } from 'carbon-components-svelte';

	const tooSmall = breakpointObserver().smallerThan('lg');
</script>
```

**Comment by metonym** [comment]
This is great, I love the idea.

Does it have to replace the existing component? I think it can exist separately.

**Comment by brunnerh** [comment]
I guess it would not hurt to have both if the core logic is extracted, thus not causing additional maintenance or potential for inconsistencies.

**Comment by metonym** [comment]
Released in [v0.61.0](https://github.com/carbon-design-system/carbon-components-svelte/releases/tag/v0.61.0)

**Comment by brunnerh** [comment]
@metonym The breaking changes release notes are missing that `breakpoints` can no longer be bound from the component and should be imported from `carbon-components-svelte/src/Breakpoint`.
